src/smtsolvers/LookaheadSMTSolver.cc

```diff
@@ -342,28 +309,47 @@ std::pair<LookaheadSMTSolver::laresult,Lit> LookaheadSMTSolver::lookaheadLoop()
         }
         if (value(v) == l_Undef)
         {
-#ifdef LADEBUG
-           printf("Updating var %d to (%d, %d)\n", v, p0, p1);
-#endif
+            // updating var score
             score->setLAValue(v, p0, p1);
             score->updateLABest(v);
         }
     }
-    Lit best = score->getBest();
+    best = score->getBest();
     if (static_cast<unsigned int>(trail.size()) == dec_vars && best == lit_Undef) {
-#ifdef LADEBUG
-        printf("All variables are already set, so we have nothing to branch on and this is a SAT answer\n");
-#endif
+        // all variables are set
         return {laresult::la_sat, best};
     }
-    assert(best != lit_Undef);
-#ifdef LADEBUG
-    printf("Lookahead phase over successfully\n");
-    printf("Best I found propagates high %d and low %d\n",
-           LAexacts[var(best)].getEx_h(),
-           LAexacts[var(best)].getEx_l());
-#endif
+
+    // lookahead phase is over
     idx = (idx + i) % nVars();
     if (!okToPartition(var(best))) { unadvised_splits++; }
     return {laresult::la_ok, best};
 }
+
+
+void LookaheadSMTSolver::cancelUntil(int level)
```

Comment:
This seems mostly like a duplicate of ``CoreSMTSolver::cancelUntil, with only one line added.

I think you could do the following, no?
```
void LookaheadSMTSolver::cancelUntil(int level) {
CoreSMTSolver::cancelUntil(level);
crossed_assumptions = min(crossed_assumptions, level);
}
```

Reply:
Fixed